src/MongoDB.Driver.Core/Core/WireProtocol/Messages/CommandMessage.cs

```diff
@@ -63,12 +64,28 @@ public sealed class CommandMessage : MongoDBMessage
         public override MongoDBMessageType MessageType => MongoDBMessageType.Command;
 
         /// <summary>
-        /// Gets a value indicating whether another message immediately follows this one.
+        /// Gets or sets a value indicating whether another message immediately follows this one with no response expected.
         /// </summary>
         /// <value>
         ///   <c>true</c> if another message immediately follows this one; otherwise, <c>false</c>.
         /// </value>
-        public bool MoreToCome => _moreToCome;
+        public bool MoreToCome
+        {
+            get { return _moreToCome; }
+            set { _moreToCome = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the delegate called to after the message has been written by the encoder.
+        /// </summary>
+        /// <value>
+        /// The post write delegate.
+        /// </value>
+        public Action<IMessageEncoderPostProcessor> PostWriteAction
```

Comment:
This new PostWriteAction will be called by the binary encoder after the message has been written.

At that time the PostWriteAction can see whether the batch was split and can change the write concern from { w : 0 } to { w : 1 } if necessary.